gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/model/GbStudentGradeInfo.java

```diff
@@ -76,4 +82,20 @@ public void addCategoryAverage(final Long categoryId, final Double score) {
 		this.categoryAverages.put(categoryId, score);
 	}
 
+	/**
+	 * Helper to convert ResourceProperties of user properties to a simple map.
+	 * @param rp
+	 * @return Map<String, String>
+	 */
+	private Map<String, String> flattenPropertiesToMap(ResourceProperties rp) {
```

Comment:
Actually, `rp.getPropertyNames` returns an Iterator which is unfortunate and makes a `stream().collect(Colllctors.toMap`... a bit more difficult.